setup.py

```diff
@@ -221,27 +221,43 @@ def run(self):
         ''' run command '''
 
         has_errors = False
+        playbooks = set()
+        included_playbooks = set()
 
         for yaml_file in find_files(
                 os.path.join(os.getcwd(), 'playbooks', 'byo'),
                 None, None, r'\.ya?ml$'):
             with open(yaml_file, 'r') as contents:
-                for line in contents:
-                    # initialize_groups.yml is used to identify entry point playbooks
-                    if re.search(r'initialize_groups\.yml', line):
-                        print('-' * 60)
-                        print('Syntax checking playbook: %s' % yaml_file)
-                        try:
-                            subprocess.check_output(
-                                ['ansible-playbook', '-i localhost,',
-                                 '--syntax-check', yaml_file]
-                            )
-                        except subprocess.CalledProcessError as cpe:
-                            print('{}Execution failed: {}{}'.format(
-                                self.FAIL, cpe, self.ENDC))
-                            has_errors = True
-                        # Break for loop, no need to continue looping lines
-                        break
+                for task in yaml.safe_load(contents.read()):
+                    if not isinstance(task, dict):
+                        # Skip yaml files which do not contain plays or includes
+                        continue
+                    if 'include' in task.keys():
+                        # Add the playbook and capture included playbooks
+                        playbooks.add(yaml_file)
+                        included_file_name = task['include'].split()[0]
+                        included_file = os.path.normpath(
+                            os.path.join(os.path.dirname(yaml_file),
+                                         included_file_name))
+                        included_playbooks.add(included_file)
+                    elif 'hosts' in task.keys():
```

Comment:
I've unraveled the mystery.  The problem was two fold and hidden by the first.  What I was actually encountering was a difference in the type of object being read from the yaml file.  In one instance ([cluster_hosts.yml](https://github.com/openshift/openshift-ansible/blob/master/playbooks/byo/openshift-cluster/cluster_hosts.yml)), I was presented with a `str` and was getting a positive on `hosts` in that `str`.  When I ultimately added a check to skip the task if it was not a `dict`, I was able to bypass that false positive.  So, I don't really need the `keys` as you suggested since I'm already verifying I'm working with a `dict`.  Updated.  :smiley: